astroquery/nasa_exoplanet_archive/core.py

```diff
@@ -0,0 +1,579 @@
+            )
+
+        if regularize:
+            object_name = self._regularize_object_name(object_name)
+
+        if "where" in criteria:
+            warnings.warn(
+                "Any filters using the 'where' argument are ignored in ``query_object``",
+                InputWarning,
+            )
+        criteria["where"] = "{0}hostname='{1}' OR {0}name='{1}'".format(prefix, object_name.strip())
+
+        return self.query_criteria_async(
+            table, get_query_payload=get_query_payload, cache=cache, **criteria,
+        )
+
+    @class_or_instance
+    def query_aliases(self, object_name, cache=None):
+        """
+        Search for aliases for a given confirmed planet or planet host
+
+        Parameters
+        ----------
+        object_name : str
+            The name of a planet or star to regularize using the ``aliastable`` table.
+        cache : bool, optional
+            Should the request result be cached? This can be useful for large repeated queries,
+            but since the data in the archive is updated regularly, this defaults to ``False``.
+
+        Returns
+        -------
+        response : list
+            A list of aliases found for the object name. The default name will be listed first.
+        """
+        return list(
+            self.query_criteria(
+                "aliastable", objname=object_name.strip(), cache=cache, format="csv"
+            )["aliasdis"]
+        )
+
+    @class_or_instance
+    def _regularize_object_name(self, object_name):
+        """Regularize the name of a planet or planet host using the ``aliastable`` table"""
+        try:
+            aliases = self.query_aliases(object_name, cache=False)
+        except RemoteServiceError:
+            aliases = []
+        if len(aliases):
+            return aliases[0]
+        warnings.warn("No aliases found for name: '{0}'".format(object_name), NoResultsWarning)
+        return object_name
+
+    def _handle_error(self, text):
+        """
+        Parse the response from a request to see if it failed
+
+        Parameters
+        ----------
+        text : str
+            The decoded body of the response.
+
+        Raises
+        ------
+        InvalidColumnError :
+            If ``select`` included an invalid column.
+        InvalidTableError :
+            If the queried ``table`` does not exist.
+        RemoteServiceError :
+            If anything else went wrong.
+        """
+        # Error messages will always be formatted starting with the word "ERROR"
+        if not text.startswith("ERROR"):
+            return
+
+        # Some errors have the form:
+        #   Error type: ...
+        #   Message: ...
+        # so we'll parse those to try to provide some reasonable feedback to the user
+        error_type = None
+        error_message = None
+        for line in text.replace("<br>", "").splitlines():
+            matches = re.findall(r"Error Type:\s(.+)$", line)
+            if len(matches):
```

Comment:
In a few places the linter gives `C1801:Do not use len(SEQUENCE) as condition value`. It seems that [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) recommends to use `if matches:` because empty sequences are false.  I am not sure if this is a reasonable recommendation and probably it is ok to ignore the rule.

Reply:
General question here: I've usually used `re.search` instead of `re.findall`.  Anyone know if there's a reason to prefer one way over another?  I'm pretty sure I just adopted that pattern a decade ago with no basis, so I'd like to learn if there's a better approach for doing boolean `re` checks like this.

Reply:
No particular reason (just what I'm used to) and I'm sure search is better. I'm happy to change it!

Reply:
No need to change it, I was just curious

Reply:
Too late :)

The API is currently offline (all requests are hanging) so I'm waiting to run the remote data tests and then I'll push the update.